app/src/org/commcare/utils/StorageUtils.java

```diff
@@ -63,9 +62,10 @@ public static int getNumIncompleteForms() {
             return new FormRecord[0];
         }
 
-        // Order ids so they're submitted to and processed by the server in
-        // the correct order.
-        sortRecordsByDate(ids, storage);
+        if (sortRequired) {
+            // Order ids so they're submitted to and processed by the server in the correct order.
+            sortRecordsBySubmissionOrderingNumber(ids, storage);
```

Comment:
Just so that in the for loop below, by reading out the records from storage in order of the sorted IDs, we're making sure that the `FormRecord[]` that this method returns is also in that same sorted order.